app/controlplane/pkg/data/workflow.go

```diff
@@ -54,9 +56,34 @@ func (r *WorkflowRepo) Create(ctx context.Context, opts *biz.WorkflowCreateOpts)
 		return nil, err
 	}
 
-	wf, err := r.data.DB.Workflow.Create().
+	// Create project and workflow in a transaction
+	tx, err := r.data.DB.Tx(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("starting transaction: %w", err)
+	}
+
+	defer func() {
+		if err != nil {
+			if rbErr := tx.Rollback(); rbErr != nil {
+				r.log.Errorf("rolling back transaction: %v", rbErr)
+			}
+		}
+	}()
+
+	// Find or create project.
+	projectID, err := tx.Project.Create().SetName(opts.Project).SetOrganizationID(orgUUID).
```

Comment:
@javirln @jiparis this is the interesting bit.

To create a workflow and a project only if it does not exist